File: src/test/java/br/com/blz/testjava/dto/ProductDtoTest.java
package br.com.blz.testjava.dto;

import br.com.blz.testjava.domain.Product;
import org.junit.Test;

import static org.junit.Assert.*;

public class ProductDtoTest {

    @Test
    public void productDtoIsMarketableFalseTest() {
        Product product = new Product();
        product.setSku(123456L);
        product.setName("Sample 1");

        InventoryDto inventoryDto = new InventoryDto();
        inventoryDto.setQuantity(0);

        ProductDto productDto = new ProductDto(product, inventoryDto);

        long skuExpected = 123456L;
        String nameExpected = "Sample 1";

        assertEquals("SKU não compatível", skuExpected, productDto.getSku().longValue());
        assertTrue("Falha ao montar o dna", nameExpected.equals(productDto.getName()));
        assertFalse("isMarketable incorreto", productDto.isMarketable());
    }

    @Test
    public void productDtoIsMarketableTrueTest() {
        Product product = new Product();
        product.setSku(123456L);
        product.setName("Sample 1");

        InventoryDto inventoryDto = new InventoryDto();
        inventoryDto.setQuantity(1);

        ProductDto productDto = new ProductDto(product, inventoryDto);

        long skuExpected = 123456L;
        String nameExpected = "Sample 1";

        assertEquals("SKU não compatível", skuExpected, productDto.getSku().longValue());
        assertTrue("Falha ao montar o dna", nameExpected.equals(productDto.getName()));
        assertTrue("isMarketable incorreto", productDto.isMarketable());
    }

}

File: src/test/kotlin/br/com/blz/testjava/controller/dto/ProductDTOTest.kt
package br.com.blz.testjava.controller.dto

import br.com.blz.testjava.domain.entities.Inventory
import br.com.blz.testjava.domain.entities.Product
import br.com.blz.testjava.domain.entities.Warehouse
import br.com.blz.testjava.domain.entities.WarehouseType
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class ProductDTOTest {

  @Test
  fun `Must convert a Product with isMarketable true`() {
    val warehouses = listOf(Warehouse("SP", 10, WarehouseType.ECOMMERCE))
    val inventory = Inventory(warehouses)
    val product = Product(123456, "Product 1", inventory)

    val productDTO = ProductDTO(product)


    assertEquals(productDTO.sku, product.sku)
    assertEquals(productDTO.name, product.name)
    assertEquals(productDTO.isMarketable, product.isMarketable())
  }

  @Test
  fun `Must convert a Product with isMarketable false`() {
    val warehouses = listOf(Warehouse("SP", 0, WarehouseType.ECOMMERCE))
    val inventory = Inventory(warehouses)
    val product = Product(123456, "Product 1", inventory)

    val productDTO = ProductDTO(product)


    assertEquals(productDTO.sku, product.sku)
    assertEquals(productDTO.name, product.name)
    assertEquals(productDTO.isMarketable, product.isMarketable())
  }

}
